### ISSUE TITLE
CRITICAL If user denied permission on Android, infinite loop 

### ISSUE THREAD

**Issue by EArminjon** [juliansteenbakker/mobile_scanner#419]
I clone the project
git checkout tags/v3.0.0-beta.3

in run the main with the barcode_scanner_controller.dart page.
I denied camera permission two times (because two popup).

Got following logs :
```
Launching lib/main.dart on sdk gphone64 arm64 in debug mode...
Running Gradle task 'assembleDebug'...
✓  Built build/app/outputs/flutter-apk/app-debug.apk.
Installing build/app/outputs/flutter-apk/app.apk...
Debug service listening on ws://127.0.0.1:64212/ecMlmRRZZZI=/ws
Syncing files to device sdk gphone64 arm64...
E/SurfaceSyncer( 8635): Failed to find sync for id=0
W/Parcel  ( 8635): Expecting binder but got null!
D/EGL_emulation( 8635): app_time_stats: avg=1130.09ms min=3.20ms max=3238.79ms count=3
D/CompatibilityChangeReporter( 8635): Compat change id reported: 78294732; UID 10168; state: ENABLED
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
D/EGL_emulation( 8635): app_time_stats: avg=60.26ms min=8.75ms max=838.03ms count=20
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
D/EGL_emulation( 8635): app_time_stats: avg=1190.36ms min=27.91ms max=2352.82ms count=2
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
D/EGL_emulation( 8635): app_time_stats: avg=361.30ms min=29.55ms max=6947.53ms count=22
D/EGL_emulation( 8635): app_time_stats: avg=24.70ms min=8.27ms max=87.32ms count=41
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
D/EGL_emulation( 8635): app_time_stats: avg=50.37ms min=39.89ms max=63.03ms count=20
D/EGL_emulation( 8635): app_time_stats: avg=23.96ms min=8.33ms max=48.01ms count=41
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
E/flutter ( 8635): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: MobileScannerException: permissionDenied
E/flutter ( 8635): #0      MobileScannerController.start (package:mobile_scanner/src/mobile_scanner_controller.dart:177:13)
E/flutter ( 8635): <asynchronous suspension>
E/flutter ( 8635): 
Application finished.
```

**Comment by juliansteenbakker** [comment]
This is fixed in #407, thanks for reporting.

**Comment by juliansteenbakker** [comment]
Please try again with v3.0.0-beta.4
